lms/static/scripts/frontend_apps/services/client-rpc.js

```diff
@@ -84,6 +87,11 @@ export class ClientRPC {
       return clientConfig;
     });
 
+    this._server.register('reportActivity', (event, data) => {
+      this.emit('annotationActivity', event, data);
+      return true;
```

Comment:
Does this result have any meaning? If we're not planning to use the result to indicate anything I'd be inclined to leave it out. We can then make a decision in `server.js` on what to send back as the `result` if the method handler didn't return anything. Otherwise there is a risk of different handlers reporting different values to indicate "no result".

Reply:
@robertknight 
See https://github.com/hypothesis/client/blob/ff711a53457cb52b5b5bb3464cbb899519065ab1/src/sidebar/util/postmessage-json-rpc.js#L71-L77

The client's Promise will reject if the response is `undefined`. "Gotta return something." Any other immediate options you want to consider?

Reply:
> The client's Promise will reject if the response is undefined. "Gotta return something." Any other immediate options you want to consider?

We could change the client code to check for the presence of the field rather than just its value, (ie. treat `{}` and `{ result: undefined }` differently) but returning `null` as the result, if the handler returns a nullish value, would I think serve the same purpose, and is what you'd expect if communicating eg. with a backend server that didn't have two different nullish types.

Reply:
> @robertknight Given that this will require changes to the client first, OK to land this as-is and circle back?

Yes, but can you add a comment indicating why the unused result is returned.